### PR TITLE
Mwsrange

### PR DESCRIPTION
**Pull request by typohnebild**
I found the mws:range in the mws code (https://github.com/MathWebSearch/mws/blob/master/src/mws/types/CmmlToken.hpp#L54) and it seems to work.
So i created two bindings for this so it is usable in queries. \range{low}{high} creates a <mws:range low="low" high="high"/> and \relrange{x}{p} creates a <mws:range low="{x-p%}" high="{x+p%}" /> as presentation-part the bindings create [low...high]

it seems this can go in the direction of that issue MathWebSearch/LaTeXML-Plugin-MathWebSearch#4

### CODE REVIEW

**Comment by tkw1536** [comment]
LGTM, however:

- We should squash all commits when this PR is merged
- We need to investigate the failing tests. It seems that Perl 5.10 doesn't seem to work.

**Comment by dginev** [comment]
As of latexml 0.8.4, perl 5.10 is no longer supported, and you should phase out those travis tests entirely. :rocket:

**Comment by tkw1536** [comment]
LGTM now. @dginev can we merge this?

File: t/2_mwsrange.t [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use LaTeXML::Util::Test;
+use File::Copy qw(copy);
+
+# Ok i'll guess this is a realy dirty workaround, but hmmm
```
Comment (dginev) [inline t/2_mwsrange.t]: Do you really want to check in this workaround? Looks quite messy...

File: t/mws_range/mws_range.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        <mrow xml:id="p1.1.m1.1.9" xref="p1.1.m1.1.9.cmml"><mo mathcolor="red" xml:id="p1.1.m1.1.9a" xref="p1.1.m1.1.9.cmml">[</mo>47.50...52.50<mo mathcolor="red" xml:id="p1.1.m1.1.9b" xref="p1.1.m1.1.9.cmml">]</mo></mrow>
+        <mo xml:id="p1.1.m1.1.11.2.1b" xref="p1.1.m1.1.11.2.1.cmml">⁢</mo>
+        <mrow xml:id="p1.1.m1.1.10" xref="p1.1.m1.1.10.cmml"><mo mathcolor="red" xml:id="p1.1.m1.1.10a" xref="p1.1.m1.1.10.cmml">[</mo>-1.00...1.00<mo mathcolor="red" xml:id="p1.1.m1.1.10b" xref="p1.1.m1.1.10.cmml">]</mo></mrow>
+      </mrow>
+    </mrow>
+    <annotation-xml xmlns:m="http://www.w3.org/1998/Math/MathML" xmlns:mws="http://search.mathweb.org/ns" encoding="MWS-Query">
+      <apply xml:id="p1.1.m1.1.11.cmml" xref="p1.1.m1.1.11">
+        <plus xml:id="p1.1.m1.1.6.cmml" xref="p1.1.m1.1.6"/>
+        <apply xml:id="p1.1.m1.1.11.1.cmml" xref="p1.1.m1.1.11.1">
+          <times xml:id="p1.1.m1.1.11.1.1.cmml" xref="p1.1.m1.1.11.1.1"/>
```
Comment (dginev) [inline t/mws_range/mws_range.xml]: Maybe this is just pragmatic for testing, but it's one strange content MathML tree - you multiply 5 ranges together? Wouldn't you prefer to have five separate formulas, each containing one range?  Just a thought.

File: t/mws_range/mws_range.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<math xmlns="http://www.w3.org/1998/Math/MathML" alttext="\range{1}{5}\range{}{5}\range{1}{}\range{}{}\range{1}{5}+\qvar@construct{x}%&#10;\range{9.00}{11.00}\range{47.50}{52.50}\range{-1.00}{1.00}" display="inline" xml:id="p1.1.m1.1">
+  <semantics xml:id="p1.1.m1.1a">
+    <mrow xml:id="p1.1.m1.1.11" xref="p1.1.m1.1.11.cmml">
+      <mrow xml:id="p1.1.m1.1.11.1" xref="p1.1.m1.1.11.1.cmml">
+        <mrow xml:id="p1.1.m1.1.1" xref="p1.1.m1.1.1.cmml"><mo mathcolor="red" xml:id="p1.1.m1.1.1a" xref="p1.1.m1.1.1.cmml">[</mo>1...5<mo mathcolor="red" xml:id="p1.1.m1.1.1b" xref="p1.1.m1.1.1.cmml">]</mo></mrow>
```
Comment (dginev) [inline t/mws_range/mws_range.xml]: Hm, the range snippets such as `1...5` ought to be in a `<mn>` element in the presentation, or some other, but definitely shouldn't free-float in an mrow

**Comment by dginev** [comment]
I didn't realize I'm to review, but now that you mention it, I left a couple of notes on the MathML used

File: t/2_mwsrange.t (hunk 1 above)
Comment (typohnebild) [reply t/2_mwsrange.t]: I found no less messy way to make the tests possible. Since latexml_tests is expecting the latemlc in the blib/scripts folder

**Comment by dginev** [comment]
Commits such as 6471f46 are a pretty good sign that the testing setup can be improved. If you need the PR merged quickly, feel free to go ahead now that Travis passes, but simplifying/enhancing the latexml test setup is definitely  something to keep in mind going forward. The root of the friction is that `LaTeXML::Util::Test` is a test module that was only ever intended to be used from within latexml, so it would need to be generalized for plugins to use it directly.

Merging :+1:

**Comment by dginev** [comment]
Oh and good work on the PR @typohnebild ! Pretty cool to see the MWS query language getting improved.

**Comment by typohnebild** [comment]
Thank you very much @dginev and thanks for the feedback

**Pull request merged**